test/surface_test.py

```diff
@@ -757,41 +814,83 @@ def test_blit__SRCALPHA_opaque_source(self):
         for pt in test_utils.rect_area_pts(src.get_rect()):
             self.assertEqual(dst.get_at(pt)[1], src.get_at(pt)[1])
 
-    def todo_test_blit__blit_to_self(self):  # TODO
-        src = pygame.Surface((256, 256), SRCALPHA, 32)
-        rect = src.get_rect()
-
-        for pt, color in test_utils.gradient(rect.width, rect.height):
-            src.set_at(pt, color)
+    def test_blit__blit_to_self(self):
+        """Test that blit operation works on self, alpha value is correct, and that no RGB distortion occurs."""
+        test_surface = pygame.Surface((128, 128), SRCALPHA, 32)
+        area = test_surface.get_rect()
 
-        src.blit(src, (0, 0))
+        for pt, test_color in test_utils.gradient(area.width, area.height):
+            test_surface.set_at(pt, test_color)
 
-    def todo_test_blit__SRCALPHA_to_SRCALPHA_non_zero(self):  # TODO
-        # " There is no unit test for blitting a SRCALPHA source with non-zero
-        #   alpha to a SRCALPHA destination with non-zero alpha " LL
+        reference_surface = test_surface.copy()
 
-        w, h = size = 32, 32
+        test_surface.blit(test_surface, (0, 0))
 
-        s = pygame.Surface(size, pygame.SRCALPHA, 32)
-        s2 = s.copy()
+        for x in range(area.width):
+            for y in range(area.height):
+                (r, g, b, a) = reference_color = reference_surface.get_at((x,y))
+                expected_color = (r, g, b, (a+(a*((256-a)//256))))
+                self.assertEqual(reference_color, expected_color)
 
-        s.fill((32, 32, 32, 111))
-        s2.fill((32, 32, 32, 31))
+        self.assertEqual(reference_surface.get_rect(), test_surface.get_rect())
 
-        s.blit(s2, (0, 0))
 
-        # TODO:
-        # what is the correct behaviour ?? should it blend? what algorithm?
+    def test_blit__SRCALPHA_to_SRCALPHA_non_zero(self):
+        """Tests blitting a nonzero alpha surface to another nonzero alpha surface
+         both straight alpha compositing method. Test is fuzzy (+/- 1/256) to account for
```

Comment:
I think this is a good way to work around the SDL1 -> SDL2 differences (which is considered a bug, but that's covered by another test).